src/main/java/com/michelin/ns4kafka/controller/UserController.java

```diff
@@ -64,4 +98,30 @@ public HttpResponse<KafkaUserResetPassword> resetPassword(String namespace, Stri
         sendEventLog(response, ApplyStatus.changed, null, response.getSpec());
         return HttpResponse.ok(response);
     }
+
+    /**
+     * Check against broker if given password matches actual one
+     *
+     * @param namespace The namespace
+     * @param user      The user
+     * @param password  The current password
+     * @return http status 200 (correct) or 401 (wrong password, thus unauthorized)
+     */
+    @Post("/{user}/check-password")
+    public HttpResponse<Void> checkPassword(String namespace, String user, @Body String password) {
+        log.debug("asked to checkPassword '{}' for user '{}'", password, user);
+        Namespace ns = getNamespace(namespace);
+
+        if (!ns.getSpec().getKafkaUser().equals(user)) {
+            throw new ResourceValidationException(KAFKA_USER_CHECK_PASSWORD, user, invalidKafkaUser(user));
+        }
+
+        UserAsyncExecutor userAsyncExecutor =
+            applicationContext.getBean(UserAsyncExecutor.class, Qualifiers.byName(ns.getMetadata().getCluster()));
+
+        if (userAsyncExecutor.checkPassword(ns.getSpec().getKafkaUser(), password)) {
```

Comment:
As suggested in https://github.com/michelin/kafkactl/issues/91, a status that specifies if the connection with the given password ends with `SUCCESS` or `FAIL` may be more significant.